### ISSUE TITLE
Resources not cleanly disposed of in the git file finder

### ISSUE THREAD

**Issue by SnarkBoojum** [pypa/setuptools-scm#403]
With 3.4.3, I still have a failure with the testsuite:

> Exception ignored in: <_io.FileIO name=8 mode='rb' closefd=True>
ResourceWarning: unclosed file <_io.BufferedReader name=8>

and I'm still "fixing" it by calling close(proc.stdout) before proc.terminate() in the finally clause.

**Comment by SnarkBoojum** [comment]
[clean_file_close.txt](https://github.com/pypa/setuptools_scm/files/4101490/clean_file_close.txt)

**Comment by RonnyPfannschmidt** [comment]
would i kill you to submit a pull request?

**Comment by blueyed** [comment]
Just for reference (from the attachment):
```diff
Description: cleanly close a file so tests don't break
Author: Julien Puydt
Forwarded: https://github.com/pypa/setuptools_scm/issues/403

--- setuptools-scm.orig/src/setuptools_scm/file_finder_git.py
+++ setuptools-scm/src/setuptools_scm/file_finder_git.py
@@ -4,6 +4,7 @@
 import logging
 from .file_finder import scm_find_files
 from .utils import trace
+from contextlib import closing
 
 log = logging.getLogger(__name__)
 
@@ -50,6 +51,7 @@
             return _git_interpret_archive(proc.stdout, toplevel)
         finally:
             # ensure we avoid resource warnings by cleaning up the process
+            proc.stdout.close()
             proc.terminate()
     except Exception:
         if proc.wait() != 0:
```

**Comment by RonnyPfannschmidt** [comment]
closed in #404
